### PR TITLE
Add duck-type tests to sieve_test.rb

### PR DESCRIPTION
**Pull request by esquinas**
Added some test to introduce duck-typing and some error handling when bad arguments are passed.

### CODE REVIEW

File: exercises/sieve/example.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 class Sieve
   attr_reader :range
   def initialize(limit)
+    begin
+      limit = limit.to_i
+    rescue NoMethodError 
+      raise ArgumentError, "Limit not convertible to an integer (#{limit})."
```
Comment (kotp) [inline exercises/sieve/example.rb]: Then use `Kernel#Integer(limit)`

File: exercises/sieve/example.rb [hunk 2]
```diff
@@ -1,6 +1,11 @@
 class Sieve
   attr_reader :range
   def initialize(limit)
+    begin
+      limit = limit.to_i
```
Comment (kotp) [inline exercises/sieve/example.rb]: `limit = Kernel#Integer(limit)`

Or simply `limit = Integer(limit)`

**Comment by kotp** [comment]
I realize that this changes the testing a little bit.  Also, you have a check for 0, but not a negative number.  You have the string with the number at the beginning, which is fine for the relaxed `.to_i` method, but not OK with the more defensive `Integer` method.

Thanks for tackling this, please do squash your commits when they are ready for review.

**Comment by kytrinyx** [comment]
Once they're ready for review, please also add a comment to the PR... otherwise we don't get notified, and the only way I know to look again is if someone triggers a notification :heart:

**Comment by esquinas** [comment]
I am sorry I am not experienced with Git, I won't know what to do next. 
I am just happy I was able to make the automated tests pass.
Really, I think the Sieve exercise is hard enough as it is without adding the _duck-typing lesson_. Which only makes sense in this case with the `.to_int` method, btw.  
I will close this pull request, but please consider adding a new exercise just for teaching duck-typing or, maybe, making it a requirement in the tests of another exercise.  
Sorry again and thanks for your comments.
Cheers, 
An exercism.io fan. :+1:

**Comment by kotp** [comment]
I think you have a point with the duck-typing part of this.  I will definitely look at this exercise though, as reviewing your changes brought up some questions.

Even if this is to remain closed, you have contributed in ways you may not have expected to, which is not uncommon!

**Pull request closed without merging**